src/sentry/static/sentry/app/components/assistant/guideAnchor.jsx

```diff
@@ -0,0 +1,81 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import React from 'react';
+import $ from 'jquery';
+
+import createReactClass from 'create-react-class';
+import Reflux from 'reflux';
+import GuideStore from '../../stores/guideStore';
+
+// A guide anchor provides a ripple-effect on an element on the page to draw attention
+// to it. Guide anchors register with the guide store, which uses this information to
+// determine which guides can be shown on the page.
+const GuideAnchor = createReactClass({
+  propTypes: {
+    target: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+  },
+
+  mixins: [Reflux.listenTo(GuideStore, 'onGuideStateChange')],
+
+  getInitialState() {
+    return {
+      active: false,
+    };
+  },
+
+  componentDidMount() {
+    GuideStore.registerAnchor(this);
```

Comment:
Ideally we don't call stores directly like this (we currently do have a lot of code that does), but ideally components perform actions through actionCreators (which creates the action), which stores listen to and update their state and then push down to components.